GameEngineTests/Players/EpsilonGreedyPlayerTests.cs

```diff
@@ -0,0 +1,61 @@
+﻿using System.Linq;
+using GameEngine;
+using GameEngine.Players;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace GameEngineTests.Players
+{
+    [TestClass]
+    public class EpsilonGreedyPlayerTests
+    {
+        private CardType[] OneCardOfEachColor
+        {
+            get
+            {
+                return new Deck(1).Cards
+                    .Where(card => card != CardType.Sun)
+                    .ToArray();
+            }
+        }
+
+        [TestMethod]
+        public void ShouldPlayCardOnOnlyOwlThatGoesFurthestWhenUsingGreedyStrategy()
+        {
+            var player = new EpsilonGreedyPlayer(1);
+            var board = new GameBoard(2, 1);
+            player.Hand.Add(OneCardOfEachColor);
+
+            var play = player.FormulatePlay(board);
+
+            Assert.AreEqual(CardType.Red, play.Card);
+            Assert.AreEqual(0, play.Position);
+        }
+
+        [TestMethod]
+        public void ShouldPlayCardOnOwlThatGoesFurthestWithHootingIntoNestWhenUsingGreedyStrategy()
+        {
+            var player = new EpsilonGreedyPlayer(1);
+            var board = new GameBoard(2);
+            board.Owls.Move(0, 6);
+            player.Hand.Add(OneCardOfEachColor);
+
+            var play = player.FormulatePlay(board);
+
+            Assert.AreEqual(CardType.Red, play.Card);
+            Assert.AreEqual(1, play.Position);
+        }
+
+        [TestMethod]
+        public void ShouldPlayCardAtRandomFromHandWhenUsingRandomStrategy()
```

Comment:
❔ It's too bad there's not a better way to assert that the random strategy was used. Is it desirable and viable to mock or spy on either of `base.FormulatePlay` or `Hand.RandomCard`? I don't think we've used any mocks in our testing yet.

Reply:
Yeah, i was thinking about how we should do this, and the best I could think of was this way (without using mocks). I could see the argument for adding mocks to assert the randomness, but I also am all for keeping our tests mock-less.

Reply:
We talked about maybe using `SeededRandom` to set up a slightly more rigorous test, but decided it wasn't worth the effort.